main.py

```diff
@@ -192,7 +192,8 @@ def normalize_brands(coll_dict):
         to_res = {k: product.pop(k) for k in both if k in product.keys() and k != "type"}
         to_res["type"] = "P"
         to_res["features"] = {k: v for k, v in product.items() if k not in item_keys or k == "type"}
-        result.append(to_res)
+        if to_res not in result:
+            result.append(to_res)
```

Comment:
```suggestion
        result.append(to_res)
```
but result should be a `set()` (or ´{}`, same thing). Sets ignore duplicates by default.

Reply:
Sets cannot contain dicts, so this can't work. Closing this conversation.